app/src/main/java/camp/nextstep/edu/github/MainViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package camp.nextstep.edu.github
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import camp.nextstep.edu.github.domain.GetGithubRepositoryUsecase
+import camp.nextstep.edu.github.domain.RepositoryVO
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(private val useCase: GetGithubRepositoryUsecase) : ViewModel() {
+
+    private val _items: MutableLiveData<List<RepositoryVO>> = MutableLiveData()
+    val items: LiveData<List<RepositoryVO>>
+        get() = _items
+
+
+    private val _errorMsg: SingleLiveEvent<ErrorEvent> = SingleLiveEvent()
+    val errorMsg: LiveData<ErrorEvent>
+        get() = _errorMsg
+
+    fun getGitHubRepository() {
+        viewModelScope.launch {
+            val result = useCase.invoke()
+            if (result.isSuccess) {
+                _items.value = result.getOrNull()
+            } else {
+                _errorMsg.value = ErrorEvent.NetworkError
+            }
```

Comment:
```suggestion
            useCase.invoke()
                .onSuccess { 
                    _items.value = it
                }
                .onFailure {
                    _errorMsg.value = ErrorEvent.NetworkError
                }
```
Result 타입에서 제공하는 확장함수를 활용하면, 좀 더 가독성있게 코드를 작성하실 수 있어요.